bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/TypeConverter.java

```diff
@@ -463,6 +469,51 @@ private static boolean isConvertibleToArrayType(Object sourceValue, BArrayType t
         return true;
     }
 
+    private static boolean isConvertibleToTupleType(Object sourceValue, BTupleType targetType,
```

Comment:
Most of the code in this function seems to be similar to the previous function `isConvertibleToArrayType`. Shall we try to reduce duplication by reusing code?

Reply:
Addressed with https://github.com/ballerina-platform/ballerina-lang/pull/31779/commits/93a63a8b643f9e14f56219f8cc2aba8a54280939